### PR TITLE
Exposing year on CRON expression and next 10 executions logic improvements

### PR DESCRIPTION
**Pull request by fwantastic**
As described in #1822, CRON expression allows year to be specified. This pull request will expose year textbox in schedule panel, allow one time execution, and properly update the next 10 executions based on the CRON expression with a slightly improved descriptions. 

Sample screenshots:
One time execution
image

0 occurrences
image

A typical CRON expression
image

### CODE REVIEW

**Comment by kunkun-tang** [comment]
Would you mind pasting some figure example show what's the change like?

**Comment by fwantastic** [comment]
@kunkun-tang they are quite large but I added screenshots for a better understanding

File: azkaban-common/src/main/java/azkaban/trigger/builtin/BasicTimeChecker.java [hunk 1]
```diff
@@ -171,7 +171,13 @@ private long calculateNextCheckTime() {
         break;
       } else if (this.cronExecutionTime != null) {
         final Date nextDate = this.cronExecutionTime.getNextValidTimeAfter(date.toDate());
-        date = new DateTime(nextDate);
+        // It's possible to have a cron expression
+        // that will schedule a finite number of executions.
```
Comment (kunkun-tang) [inline azkaban-common/src/main/java/azkaban/trigger/builtin/BasicTimeChecker.java]: don't quite understand this comment. Do you want to say "Some Cron Expressions possibly do not have follow-up occurrences"?

Comment (fwantastic) [reply azkaban-common/src/main/java/azkaban/trigger/builtin/BasicTimeChecker.java]: @kunkun-tang That is correct. Would you want me to update that comment?

**Pull request merged**